espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/Http/Server.java

```diff
@@ -3,35 +3,41 @@
 import com.google.gson.GsonBuilder;
 
 import java.io.IOException;
+import java.io.PrintWriter;
+import java.io.StringWriter;
 
 import fi.iki.elonen.NanoHTTPD;
 import io.appium.espressoserver.lib.Exceptions.DuplicateRouteException;
-import io.appium.espressoserver.lib.Exceptions.ServerErrorException;
-import io.appium.espressoserver.lib.Http.Response.BaseResponse;
+import io.appium.espressoserver.lib.Http.Response.AppiumResponse;
+import io.appium.espressoserver.lib.Http.Response.ErrorResponse;
 import io.appium.espressoserver.lib.Model.AppiumStatus;
 import io.appium.espressoserver.lib.Model.GsonAdapters.AppiumStatusAdapter;
 
 public class Server extends NanoHTTPD {
 
     private Router router;
 
-    public Server() throws IOException, ServerErrorException {
+    public Server() throws IOException, DuplicateRouteException {
         super(8080);
         start(NanoHTTPD.SOCKET_READ_TIMEOUT, false);
         System.out.println("\nRunning! Point your browsers to http://localhost:8080/ \n");
-        try {
-            router = new Router();
-        } catch (DuplicateRouteException e) {
-            throw new ServerErrorException();
-        }
+        router = new Router();
     }
 
     @Override
     public Response serve(IHTTPSession session) {
-        BaseResponse response = router.route(session);
-        GsonBuilder gsonBuilder = new GsonBuilder();
+        GsonBuilder gsonBuilder = new GsonBuilder().serializeNulls();
+        try {
+            AppiumResponse response = router.route(session);
         gsonBuilder.registerTypeAdapter(AppiumStatus.class, new AppiumStatusAdapter());
-        // TODO: Don't hardcode application/json change it to MediaType http://docs.oracle.com/javaee/6/api/javax/ws/rs/core/MediaType.html
-        return newFixedLengthResponse(response.getHttpStatus(),  "application/json", gsonBuilder.create().toJson(response.getResponse()));
+            // TODO: Don't hardcode application/json change it to MediaType http://docs.oracle.com/javaee/6/api/javax/ws/rs/core/MediaType.html
+            return newFixedLengthResponse(response.getHttpStatus(), "application/json", gsonBuilder.create().toJson(response));
+        } catch (Exception e) {
+            StringWriter sw = new StringWriter();
```

Comment:
maybe it worth to extract it into a separate utility method. Might be useful elsewhere.
Also, you may use https://commons.apache.org/proper/commons-lang/javadocs/api-3.1/org/apache/commons/lang3/exception/ExceptionUtils.html#getStackTrace(java.lang.Throwable) for the same purpose